src/sdk/object_sdk.js

```diff
@@ -207,19 +207,16 @@ class ObjectSDK {
     async load_requesting_account(req) {
         try {
             const token = this.get_auth_token();
-            if (!token) return;
+            if (!token && !process.env.NC_NSFS_NO_DB_ENV) return;
             this.requesting_account = await account_cache.get_with_cache({
                 bucketspace: this._get_bucketspace(),
-                access_key: token.access_key,
+                access_key: token ? token.access_key : config.ANONYMOUS_ACCESS_KEY,
             });
             if (this.requesting_account?.nsfs_account_config?.distinguished_name) {
                 const distinguished_name = this.requesting_account.nsfs_account_config.distinguished_name.unwrap();
-                const user = await dn_cache.get_with_cache({
-                    bucketspace: this._get_bucketspace(),
-                    distinguished_name,
-                });
-                this.requesting_account.nsfs_account_config.uid = user.uid;
-                this.requesting_account.nsfs_account_config.gid = user.gid;
+                const account_config = await this.get_distinguished_user_details(distinguished_name);
+                this.requesting_account.nsfs_account_config.uid = account_config.nsfs_account_config.uid;
+                this.requesting_account.nsfs_account_config.gid = account_config.nsfs_account_config.gid;
```

Comment:
this change seems to be unneeded now right? it feels a bit weird that the get_distinguished_user_details function returns a pseudo user structure when it actually only returns the uid and gid, so if not needed I would revert it.